custom_model_runner/datarobot_drum/drum/args_parser.py

```diff
@@ -863,6 +877,84 @@ def get_arg_option(options: argparse.Namespace, arg_option: str):
         value_in_namespace = arg_option.lstrip("-").replace("-", "_")
         return getattr(options, value_in_namespace, None)
 
+    @staticmethod
+    def _is_valid_iso8601(arg):
+        try:
+            datetime.fromisoformat(arg.replace("Z", "+00:00"))
+            return arg
+        except ValueError:
+            raise argparse.ArgumentTypeError(f"The value {arg} is not a valid ISO-8601 timestamp!")
+
+    @staticmethod
+    def _reg_arg_time_series(*parsers):
+        def require_both_dates(arg):
+            # First check if this is score command
+            if ArgumentsOptions.SCORE not in sys.argv[1]:
```

Comment:
Is this commonly how we do this type of argument validation in DRUM, e.g. we manually parse through `sys.argv` looking for other options? I see you doing this below but this will break down if there is a mixture of things set as env vars and things set and CLI arguments right?

If this is the existing pattern then so-be-it but if this is the first time we have such interdependencies may I suggest instead to let the args all get parsed (and only include validation that is local to the specific flag, such as parsing ISO-8601 format) and then `verify_options()` is the place to make sure the set of options as a whole are logically consistent?

Reply:
Also, I think it breaks down if users do stuff like `--myflag=12356` vs `--myflag 123456` where I think Python's argparse will accept both styles equally? So yeah, I think it is always best to do pure argument parsing in one phase and then logical validation in a second phase.